### PR TITLE
Repair isDir flag

### PR DESCRIPTION
**Pull request by mako321**

### CODE REVIEW

File: src/Http/Services/FileService.php [hunk 1]
```diff
@@ -91,13 +91,13 @@ public function listInfo(string $directory): Collection
...
             return collect($this->disk->listContents($directory, false))
-                ->filter(fn (array $metadata) => $this->checkUserAccessToFile($user, $metadata['path']))
+                ->filter(fn (array $metadata) => $this->checkUserAccessToFile($user, $metadata))
                 ->map(fn (array $metadata) => [
                     'name' => $metadata['basename'],
                     'created_at' => isset($metadata['timestamp']) ? date(DATE_RFC3339, $metadata['timestamp']) : null,
                     'mime' => $this->disk->mimeType($metadata['path']),
                     'url' => $this->disk->url($metadata['path']),
-                    'isDir' => $this->disk->mimeType($metadata['path']) === 'directory'
+                    'isDir' => $metadata['type'] === 'dir',
```
Comment (HerbertIV) [inline src/Http/Services/FileService.php:100]:
```suggestion
                    'isDir' => isset($metadata['type']) && $metadata['type'] === 'dir',
```

File: src/Http/Services/FileService.php [hunk 2]
```diff
@@ -128,7 +128,7 @@ public function findByName(string $directory, string $name): Collection
                     'url' =>  $this->disk->url($metadata['path']),
                     'created_at' => isset($metadata['timestamp']) ? date(DATE_RFC3339, $metadata['timestamp']) : null,
                     'mime' => $this->disk->mimeType($metadata['path']),
-                    'isDir' => $this->disk->mimeType($metadata['path']) === 'directory'
+                    'isDir' => $metadata['type'] === 'dir',
```
Comment (HerbertIV) [inline src/Http/Services/FileService.php:131]: Up

File: src/Http/Services/FileService.php [hunk 3]
```diff
@@ -233,20 +233,18 @@ private function saveUserAccessToDirectory(User $user, array $directories): void
         ], $user->getKey());
     }
 
-    private function checkUserAccessToFile(User $user, string $path): bool
+    private function checkUserAccessToFile(User $user, array $metadata): bool
```
Comment (HerbertIV) [inline src/Http/Services/FileService.php:236]: Do you have a 100% confidence that key in array exists ?

**Review by HerbertIV — APPROVED** [review]

**Review by qunabu — APPROVED** [review]

**Review by qunabu — CHANGES_REQUESTED** [review]
php8 test are failing

**Pull request merged**